src/main/java/com/redhat/devtools/lsp4ij/settings/ui/EditJsonSchemaDialog.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij.settings.ui;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.util.ui.FormBuilder;
+import com.redhat.devtools.lsp4ij.LanguageServerBundle;
+import com.redhat.devtools.lsp4ij.internal.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+
+/**
+ * Dialog to edit Json Schema.
+ */
+public class EditJsonSchemaDialog extends DialogWrapper {
+
+
+    private final @NotNull Project project;
+    private String jsonSchemaContent;
+    private JsonTextField jsonSchemaWidget;
+
+    protected EditJsonSchemaDialog(@NotNull Project project, String jsonSchemaContent) {
+        super(true);
+        this.project = project;
+        this.jsonSchemaContent = jsonSchemaContent;
+        super.setTitle(LanguageServerBundle.message("edit.json.schema.dialog.title"));
+        init();
+    }
+
+    @Override
+    protected @Nullable JComponent createCenterPanel() {
+        FormBuilder builder = FormBuilder
+                .createFormBuilder();
+
+        jsonSchemaWidget = new JsonTextField(project);
+        builder.addLabeledComponentFillVertically(LanguageServerBundle.message("edit.json.schema.dialog.schema"), jsonSchemaWidget);
+        if (StringUtils.isNotBlank(jsonSchemaContent)) {
+            jsonSchemaWidget.setText(jsonSchemaContent);
+        }
+
+        JPanel panel = new JPanel(new BorderLayout());
+        panel.add(builder.getPanel(), BorderLayout.CENTER);
+        return panel;
+    }
+
+    @Override
+    public void show() {

```

Comment:
@SCWells72 when dialog is shown there is a problem with size of the dialog:

 * when there is no Json Schema content, the size is too small
 * where there is a Json Schema content, the size takes the full screen.

Do you know how to fix that?

Reply:
@angelozerr the simplest way is to set a consistent size for the dialog. Here's an example of how I do it in one place that takes into account the screen size:

```java
JComponent contentPanel = getContentPanel();
Rectangle screenRectangle = contentPanel != null ? ScreenUtil.getScreenRectangle(contentPanel) : null;
int formPanelHeight = (screenRectangle != null) && (screenRectangle.height > 800) ? 700 : 600;
Dimension formPanelSize = new Dimension(1100, formPanelHeight);
formPanel.setPreferredSize(formPanelSize);
formPanel.setMinimumSize(formPanelSize);
```

In this case, `formPanel` is what's returned by `DialogWrapper#createCenterPanel()`.

Reply:
Obviously choose whatever width/height values you want. Those are tuned for the contents of a very specific dialog.

Reply:
I have hard coded the size.